server/lib/chat/chat-service.test.ts

```diff
@@ -270,27 +294,20 @@ describe('chat/chat-service', () => {
         { type: ServerChatMessageType.JoinChannel },
         dbClient,
       )
+
+      await transactionPromise.resolve()
```

Comment:
`Deferred.resolve` is not async so this await is very strange. I think you probably *do* need to `await` a promise to ensure that async things are completed, but you should split that:

```ts
transactionPromise.resolve()
await Promise.resolve()
```